### PR TITLE
Add saving/loading scheduler state_dict

### PR DESCRIPTION
**Pull request by hyperzy**
- Add saving/loading scheduler state_dict to accommodate the training resumption on some methods with lr schedulers.

### CODE REVIEW

**Review by tancik — APPROVED** [review]
Thanks for adding
